rita_common/src/rita_loop/write_to_disk.rs

```diff
@@ -0,0 +1,151 @@
+    RitaExitSettingsStruct(RitaExitSettingsStruct),
+}
+/// This loop attempts to perform all write operations for writing to disk
+/// This includes writing config/settings, usage tracker, and debt tracker.
+/// It takes in a settings enum in order to identify what type the device
+/// is.
+pub fn save_to_disk_loop(mut old_settings: SettingsOnDisk, file_path: &str) {
+    let file_path = file_path.to_string();
+    let router_storage_small;
+    let mut saving_to_disk_frequency: Duration;
+
+    match old_settings.clone() {
+        SettingsOnDisk::RitaClientSettings(old_settings_client) => {
+            match old_settings_client.network.device.clone() {
+                Some(val) => router_storage_small = is_router_storage_small(&val),
+                None => router_storage_small = false,
+            };
+            saving_to_disk_frequency = Duration::from_secs(old_settings_client.save_interval);
+        }
+        SettingsOnDisk::RitaExitSettingsStruct(old_settings_exit) => {
+            match old_settings_exit.network.device.clone() {
+                Some(val) => router_storage_small = is_router_storage_small(&val),
+                None => router_storage_small = false,
+            };
+            saving_to_disk_frequency = Duration::from_secs(old_settings_exit.save_interval);
+        }
+    }
+
+    thread::spawn(move || loop {
+        let start = Instant::now();
+
+        if start.elapsed() < saving_to_disk_frequency {
+            thread::sleep(saving_to_disk_frequency - start.elapsed());
+        }
+
+        //settings
+        match old_settings.clone() {
+            SettingsOnDisk::RitaClientSettings(old_settings_client) => {
+                let new_settings = get_rita_client();
+
+                if old_settings_client != new_settings {
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    log::info!("Config/settings is being saved to");
+                }
+
+                old_settings = SettingsOnDisk::RitaClientSettings(new_settings);
+            }
+            SettingsOnDisk::RitaExitSettingsStruct(old_settings_exit) => {
+                let new_settings = get_rita_exit();
+                saving_to_disk_frequency = Duration::from_secs(new_settings.save_interval);
+
+                if old_settings_exit != new_settings {
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    log::info!("Config/settings is being saved to");
+                }
+
+                old_settings = SettingsOnDisk::RitaExitSettingsStruct(new_settings);
+            }
+        }
+
+        // debt keeper
+        if !router_storage_small {
+            let runner = AsyncSystem::new();
+            runner.block_on(async move {
+                if let Err(e) = send_debt_update() {
```

Comment:
call the save function *only* there's no need to send a debt update, while that triggers saving right now if enough time has elapsed the point of this pr is to move that timer and saving handling into this thread and not call it from there.